internal/resources/cluster/resource_cluster.go

```diff
@@ -39,6 +49,7 @@ func ResourceTMCCluster() *schema.Resource {
 		UpdateContext: resourceClusterInPlaceUpdate,
 		DeleteContext: resourceClusterDelete,
 		Schema:        clusterSchema,
+		CustomizeDiff: ValidateKubeConfig,
```

Comment:
adding schema validation in customiseDiff is wrong way. 
From what I understand, customiseDiff is used to show the diffs and override the diffs shown or not.
Can you check the function definition in framework ? I might be wrong here

Reply:
Ref- https://github.com/hashicorp/terraform/pull/14887#issue-231839744

Usecase of `CustomizeDiff` from the terraform org: 

> Vetoing changes to the diff that could not be reasonably carried out (ie: illegal AWS RDS operations) by kicking back an error to the CustomizeDiff function, much like the other CRUD functions.

Reply:
As per the above use case, we can use `CustomizeDiff` for validations.